models/taskRequests.js

```diff
@@ -0,0 +1,119 @@
+const { TASK_REQUEST_STATUS } = require("../constants/taskRequests");
+const { TASK_STATUS } = require("../constants/tasks");
+const firestore = require("../utils/firestore");
+const taskRequestsCollection = firestore.collection("taskRequests");
+const tasksModel = require("./tasks");
+const userModel = require("./users");
+
+/**
+ * Fetch all task requests
+ *
+ * @return {Object}
+ */
+const fetchTaskRequests = async () => {
+  const taskRequests = [];
+
+  const visitedUsersId = new Set();
+  const visitedUsers = [];
+
+  const taskRequestsSnapshot = (await taskRequestsCollection.get()).docs;
+
+  for (const taskRequestSnapshot of taskRequestsSnapshot) {
+    const taskRequest = taskRequestSnapshot.data();
+    const { taskData } = await tasksModel.fetchTask(taskRequest.taskId);
+    const users = await Promise.all(
+      taskRequest.requestors.map((requestor) => {
+        if (visitedUsers.includes(requestor)) {
+          return visitedUsersId.find((visitedUserId) => (visitedUserId.id = requestor));
```

Comment:
The bug is that the visitedUsersId variable is not being initialized. This means that the find() method will always return undefined. To fix this bug, you need to initialize the visitedUsersId variable to an empty array. You can do this by adding the following line to the top of the code:

Reply:
visitedUsersId is already defined as a `Set`, thanks for pointing this out though as I need to use `.has` method instead of `.includes`.